### PR TITLE
feat: Kava Protonet Data Reset Chain CI

### PR DESCRIPTION
**Pull request by sesheffield**
## description

* handle protonet flow with new ansible playbook for reset, restart, seed
* Using different tags and using autoscaling group for ordering (want the validator to be reset first then api node)
  * Added `autoscaling groups` to be ran for `kava-node-updater` since new infra doesn't use `KavaChainId` and uses `ChainId` and this is to ensure we reset the validator node first then any api nodes afterwards

kava protonet genesis contains all the necessary fields set

This flow will be removed eventually for testnetV2

## Checklist
 - [ ] Changelog has been updated as necessary.

### CODE REVIEW

File: ci/env/kava-protonet/genesis.json [hunk 1]
```diff
@@ -235,7 +235,7 @@
     "evm": {
       "accounts": [],
       "params": {
-        "evm_denom": "aphoton",
+        "evm_denom": "ukava",
```
Comment (karzak) [inline ci/env/kava-protonet/genesis.json:238]: should be `akava`

File: ci/env/kava-protonet/genesis.json [hunk 2]
```diff
@@ -253,12 +253,12 @@
...
-          "cancun_block": "0"
+          "london_block": null,
+          "arrow_glacier_block": null,
+          "gray_glacier_block": null,
+          "merge_netsplit_block": null,
+          "shanghai_block": null,
+          "cancun_block": null
         },
         "eip712_allowed_msgs": [],
         "allow_unprotected_txs": false
```
Comment (karzak) [inline ci/env/kava-protonet/genesis.json:264]: See a bunch of other values that should be copied over from https://github.com/Kava-Labs/kava/blob/master/ci/env/kava-internal-testnet/genesis.json

ie [consensus params](https://github.com/Kava-Labs/kava/blob/ee4fc56f013c06a093f3b5a1202652d7e1e81124/ci/env/kava-internal-testnet/genesis.json#L5)

Can do a diff of different sections and defer to the internal testnet values generally

File: .github/workflows/cd-protonet.yml [hunk 3]
```diff
@@ -15,28 +17,18 @@ jobs:
...
       aws-region: us-east-1
+      chain-tag-name: ChainId
       chain-id: proto_2221-17000
-      ssm-document-name: kava-testnet-internal-node-update
-      playbook-name: reset-protonet-playbook.yml
-      playbook-infrastructure-branch: master
+      auto_scaling_group_names: kava-protonet-iavl-v1-validator-api-node #,kava-protonet-iavl-v1-rpc-api-node
+      ssm-document-name: kava-protonet-iavl-v1-instance-update
+      playbook-name: reset-chain.yml
+      playbook-infrastructure-branch: shef-protonet-validator-network-ephemeral
```
Comment (sesheffield) [inline .github/workflows/cd-protonet.yml:29]: change back to master

File: .github/workflows/cd-protonet.yml [hunk 4]
```diff
@@ -15,28 +17,18 @@ jobs:
...
+#    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    uses: ./.github/workflows/cd-reset-protonet.yml
     with:
       aws-region: us-east-1
+      chain-tag-name: ChainId
       chain-id: proto_2221-17000
-      ssm-document-name: kava-testnet-internal-node-update
-      playbook-name: reset-protonet-playbook.yml
-      playbook-infrastructure-branch: master
+      auto_scaling_group_names: kava-protonet-iavl-v1-validator-api-node #,kava-protonet-iavl-v1-rpc-api-node
```
Comment (sesheffield) [inline .github/workflows/cd-protonet.yml:26]: remove comment

File: .github/workflows/cd-protonet.yml [hunk 5]
```diff
@@ -5,6 +5,8 @@ name: Continuous Deployment (Protonet)
 #    workflows: [Continuous Integration (Kava Master)]
 #    types:
 #      - completed
+on:
```
Comment (sesheffield) [inline .github/workflows/cd-protonet.yml:8]: undo comment above

File: .github/workflows/cd-reset-protonet.yml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          AWS_REGION: ${{ inputs.aws-region }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.CI_AWS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CI_AWS_KEY_SECRET }}
+      - name: checkout infrastructure repo
+        uses: actions/checkout@v4
+        with:
+          repository: Kava-Labs/infrastructure
+          token: ${{ secrets.KAVA_PRIVATE_GITHUB_ACCESS_TOKEN }}
+          path: infrastructure
+          ref: shef-protonet-validator-network-ephemeral
```
Comment (sesheffield) [inline .github/workflows/cd-reset-protonet.yml:64]: change back to master

File: ci/env/kava-protonet/KAVA.VERSION [hunk 7]
```diff
@@ -1 +1 @@
-shef-protonet-validator-network
+shef-protonet-validator-network-ci
```
Comment (sesheffield) [inline ci/env/kava-protonet/KAVA.VERSION:1]: change to master

File: .github/workflows/cd-protonet.yml [hunk 8]
```diff
@@ -5,38 +5,31 @@ name: Continuous Deployment (Protonet)
...
   # download updated binary and genesis
-  # reset application database state (only done on internal testnet)
+  # reset application database state
+  # start kava and doctor
+  # enter inService
   reset-chain-to-zero-state:
     # only start cd pipeline if last ci run was successful
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
-    uses: ./.github/workflows/cd-reset-internal-testnet.yml
+#    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```
Comment (sesheffield) [inline .github/workflows/cd-protonet.yml:21]: remove comment

File: .github/workflows/cd-reset-internal-testnet.yml [hunk 9]
```diff
@@ -40,6 +40,7 @@ jobs:
       - name: take the chain offline
         run: bash ${GITHUB_WORKSPACE}/.github/scripts/put-all-chain-nodes-on-standby.sh
         env:
+          CHAIN_TAG_NAME: ${{ inputs.chain-tag-name }}
```
Comment (pirtleshell) [inline .github/workflows/cd-reset-internal-testnet.yml:43]: thanks for making this more configurable for the different approaches. can we give it a clearer name? maybe `AWS_CHAIN_ID_TAG_NAME`

**Review by pirtleshell — COMMENTED** [review]
does a rebase onto master resolve the failing CI?

we probably shouldn't be updating kvtool in this PR

**Comment by sesheffield** [comment]
> does a rebase onto master resolve the failing CI?
> 
> we probably shouldn't be updating kvtool in this PR

Nick's branch `nd-implement-basic-precompile-calling` updates the kvtool submodule commit to the 86012 hash.

which then causes issues for ansible with:
```
skipping: [localhost] TASK [chain-binaries : checkout chain repository] ******************************
2024-09-24T23:08:39.185Z
fatal: [localhost]: FAILED! => 
{
    "changed": false,
    "cmd": [
        "/usr/bin/git",
        "fetch",
        "--tags",
        "--force",
        "origin"
    ],
    "msg": "Failed to download remote objects and refs:  From https://github.com/kava-labs/kava\n   ac1bb5c9..69964888  dl-eip712-domainsep-tests -> origin/dl-eip712-domainsep-tests\n * [new branch]        dl-usdt-gas            -> origin/dl-usdt-gas\n * [new branch]        lbayas/genesis-updates -> origin/lbayas/genesis-updates\n   8ac4c5d0..ee4fc56f  master                 -> origin/master\n * [new branch]        nd-implement-basic-precompile-calling -> origin/nd-implement-basic-precompile-calling\n * [new branch]        shef-protonet-validator-network -> origin/shef-protonet-validator-network\n * [new branch]        shef-protonet-validator-network-ci -> origin/shef-protonet-validator-network-ci\n * [new branch]        shef-protonet-validator-network-ci-v2 -> origin/shef-protonet-validator-network-ci-v2\n   50418c2e..e2217257  ss-protonet-tester     -> origin/ss-protonet-tester\nFetching submodule tests/e2e/kvtool\nfatal: remote error: upload-pack: not our ref 86012e65f20488fa89d509c0183ac259ce55ffca\nErrors during submodule fetch:\n\ttests/e2e/kvtool\n"
}
```

but you're right, ideally we don't want any kvtool changes here, but this was the branch I needed to test since it's not in master yet.

**Comment by sesheffield** [comment]
setting to draft to triple check quality

**Pull request closed without merging**